vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt

```diff
@@ -62,17 +63,22 @@ class OidcSiopWallet(
      * Implementations need to fetch the url passed in, and return either the body, if there is one,
      * or the HTTP header `Location`, i.e. if the server sends the request object as a redirect.
      */
-    private val remoteResourceRetriever: RemoteResourceRetrieverFunction = { null },
+    private val remoteResourceRetriever: RemoteResourceRetrieverFunction? = null,
```

Comment:
Whoops, I recently developed a dislike for non-null default values, but I didn't mean for it to jump into action at work

The reason is something like: Assuming you delegate the parameter specification for a function call/object construction to somewhere else, but this somewhere else would like to use the default value. Having it like it was before would need the default value to be copied over to each delegated specification. Now, when the default value changes, this would also mean all users of the library which delegated parameter specification would need to change their default values as well if the intention was to use the default value.

Take `OidcSiopWallet.clientId` for example. When the default value for the constructor is changed at some point, it would need to be changed at least at the following two locations: primary constructor, `newDefaultInstance`. 

Should I change it back or does this actually seam reasonable?

Reply:
That is indeed a valid concern! To be honest, the `newDefaultInstance` methods were not an ideal solution from the start, because providing a second constructor (or using one with several parameters) is probably cleaner. So reconsidering this, I'm fine with the `null` default value, thanks for your explanation!